src/Microsoft.AspNetCore.Server.IIS/HttpContextExtensions.cs

```diff
@@ -34,5 +34,17 @@ public static string GetIISServerVariable(this HttpContext context, string varia
 
             return feature[variableName];
         }
+
+        /// <summary>
+        /// Sets the value of a server variable for the current request.
+        /// </summary>
+        /// <param name="context">The http context for the request.</param>
+        /// <param name="variableName">The name of the variable.</param>
+        /// <param name="variableValue">The new value for the variable.</param>
+        public static void SetIISServerVariable(this HttpContext context, string variableName, string variableValue)
+        {
+            var feature = context.Features.Get<IServerVariablesFeature>();
```

Comment:
Null check and throw InvalidOperationException, the feature could have been removed.

Reply:
I was thinking about removing these methods all together.

Reply:
Get is common enough, but I agree Set is uncommon.

Reply:
What about removing both and changing IServerVariableFeature to have methods instead of indexer?

Reply:
What's wrong with the indexer?

Reply:
It's a bit strange seeing an interface with only member being string indexer.

Reply:
*shrug* it fits the requirements.